### PR TITLE
Add: CFEngine config-files layer

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

**Comment by jimis** [comment]
dammit nick

**Comment by nickanderson** [comment]
you love it

File: layers/+config-files/cfengine/packages.el [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      '(
+	flycheck
+	company
+	;; cfengine3-mode ;; This mode is included in the default emacs distro
+	))
+
+(defun cfengine/init-cfengine()
+  (use-package cfengine3-mode
+    :init
+      (add-hook 'before-save-hook 'whitespace-cleanup)
```
Comment (nixmaniack) [inline layers/+config-files/cfengine/packages.el]: This should be left as user preference as Spacemacs already provides this configuration via `dotspacemacs-whitespace-cleanup`

Comment (nixmaniack) [inline layers/+config-files/cfengine/packages.el]: The `:init` block of `use-package` should be wrapped in `progn` as per Spacemacs convention when there are more than one statements.

File: layers/+config-files/cfengine/packages.el [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      (customize-set-variable 'cfengine-parameters-indent (quote (promise pname 2)))
+      (progn
+        (spacemacs/set-leader-keys-for-major-mode 'cfengine3-mode
+          "j" 'cfengine3-reformat-json-string
+          ))))
```
Comment (nixmaniack) [inline layers/+config-files/cfengine/packages.el]: Closing parens needn't go on newline.

**Comment by nickanderson** [comment]
Thanks for the review, I will work on it.

File: layers/+config-files/cfengine/packages.el (hunk 1 above)
Comment (nickanderson) [inline layers/+config-files/cfengine/packages.el]: ack updated to wrap in progn, moved whitespace cleanup note into readme

File: layers/+config-files/cfengine/packages.el (hunk 2 above)
Comment (nickanderson) [inline layers/+config-files/cfengine/packages.el]: ack

**Comment by nickanderson** [comment]
updated, rebased and squashed

**Comment by syl20bnr** [comment]
Thank you ! 👍 
I reformat the README to better fit the other the format and style of the other READMEs and I fixed various things in the layer, related commit: https://github.com/syl20bnr/spacemacs/commit/f3639da
I did not cherry-pick the screenshot since they are showing basic stuff which are the same in lot of other layers, we cannot include such screenshots everywhere, moreover screenshots are not easy to maintain since the look of spacemacs can change and it is hard to update them.

Cherry-picked into develop branch, you can safely delete your branch.

**Comment by nickanderson** [comment]
Thanks!

**Pull request closed without merging**